experimental/dds/tree2/src/test/shared-tree/fuzz/fuzzEditGenerators.ts

```diff
@@ -247,6 +249,45 @@ export const makeEditGenerator = (
 		}
 	};
 
+	const move = (state: FuzzTestState): FieldEditTypes => {
+		const tree = state.client.channel;
+		const fieldInfo = selectTreeField(
+			fuzzViewFromTree(tree),
+			state.random,
+			weights.fieldSelection,
+			(f) => f.type === "sequence" && f.content.length > 1,
+		);
+		assert(fieldInfo.type === "sequence", "Move should only be performed on sequence fields");
+		const { content: field } = fieldInfo;
+		assert(field.length > 1, "Sequence must have at least two elements to perform a move");
+
+		const start = state.random.integer(0, field.length - 1);
+		// If we choose the first node in the sequence we can't grab the whole sequence
+		// and still have something to move to
+		const count =
+			start === 0
+				? state.random.integer(1, field.length - 1)
+				: state.random.integer(1, field.length - start);
+
+		const validMoveIndices: number[] = [];
+		for (let i = 0; i < field.length; i++) {
```

Comment:
this should be doable in O(1) instead of O(length), but probably not a big deal as current impl is more clear. Maybe leave a comment here in case we find ourselves with too slow fuzz tests in the future.